central/reports/scheduler/schedule_test.go

```diff
@@ -0,0 +1,96 @@
+package scheduler
+
+import (
+	"bytes"
+	"testing"
+	"text/template"
+	"time"
+
+	"github.com/stackrox/rox/pkg/branding"
+	"github.com/stackrox/rox/pkg/fixtures"
+	"github.com/stackrox/rox/pkg/testutils/envisolator"
+	"github.com/stretchr/testify/assert"
+)
+
+var expectedVulnReportEmailTemplateRhacsBrandingWithPlaceholders = `
+	Red Hat Advanced Cluster Security for Kubernetes has found vulnerabilities associated with the running container images owned by your organization. Please review the attached vulnerability report {{.WhichVulns}} for {{.DateStr}}.
+
+	To address these findings, please review the impacted software packages in the container images running within deployments you are responsible for and update them to a version containing the fix, if one is available.`
+
+var expectedVulnReportEmailTemplateStackroxBrandingWithPlaceholders = `
+	StackRox has found vulnerabilities associated with the running container images owned by your organization. Please review the attached vulnerability report {{.WhichVulns}} for {{.DateStr}}.
+
+	To address these findings, please review the impacted software packages in the container images running within deployments you are responsible for and update them to a version containing the fix, if one is available.`
+
+var expectedNoVulnsFoundEmailTemplateRhacsBranding = `
+	Red Hat Advanced Cluster Security for Kubernetes has found zero vulnerabilities associated with the running container images owned by your organization.`
+
+var expectedNoVulnsFoundEmailTemplateStackroxBranding = `
+	StackRox has found zero vulnerabilities associated with the running container images owned by your organization.`
+
+type vulnsAndDate struct {
+	WhichVulns string
+	DateStr    string
+}
+
+func generateExpectedVulnReportEmailTemplates(t *testing.T) (string, string) {
+	data := &vulnsAndDate{
+		WhichVulns: "for all vulnerabilities",
+		DateStr:    time.Now().Format("January 02, 2006"),
```

Comment:
The test will flake with error if this function is called just before the midnight, and `formatMessage()` is called just after. Not a big deal, but if there's a way to not have this situation, it is better.

Reply:
Then I have two options:

1. Only compare a substring of the email since branding is right at the start.
2. Assuming we have some testing utility to control time; use that to stop time when starting the test and resume afterwards.

Now I think option 2 is much more elegant but I don't know if we have such a testing utility. @msugakov thoughts?

Reply:
Couple more options:
3. You could make `reportDate` to be an argument of the `formatMessage`.
4. You can use regular expressions to match the rest of the string but just some placeholder instead of the date. `assert` should have regex checks.

Reply:
I still think (2) would have been the most elegant but I assume we don't have such a testing utility then(?)

In which case I'll do (3) since it's the easiest and still compares the entire string.

Reply:
I don't know about (2) so would need to research it. I think I've seen it somewhere but can't recall where.
Fine with the current approach.